Add patch for sturgeon and downgrade skia

MagneFire opened the pull request:
This pull request includes the patch file needed to compile the system tarball for sturgeon. I also needed to downgrade skia to fix some deprecation errors.

I know that the patch file is huge ;). It meanly consists of the original patches used from the Lollipop build. I had to remove the directories: MultiDexLegacyAndException/ MultiDexLegacyTestApp/ MultiDexLegacyVersionedTestApp_v3/ MultiDexLegacyVersionedTestApp_v2/ MultiDexLegacyVersionedTestApp_v1/ in frameworks/base/core/tests/hosttests/test-apps/ in frameworks/base/core/tests/hosttests/test-apps/. Otherwise it wouldn't let me compile the projects. This makes the patch file huge...

I didn't really want to mess with the xml file but the instructions for the lenok platform doesn't seem to work either. The patch file is not specific for sturgeon, so this patch may also work for lenok. I just don't have a lenok watch to test this...

FlorentRevest commented:
I would prefer if we could replace the affected repositories with repos under the AsteroidOS github organization rather than adding a global patch that affects the whole android source tree and does everything at once. Your patch has no description and touches various subsystems for various reasons. It would be better if this could be split across different commits, each affecting one repository for one good reason :) 
PS: if you need new repos under the AsteroidOS organizations, just let me know which ones and I'll create them in a way that you can send PR to each of them

MagneFire commented:
Yes, I completely agree with you. Given that the version for Lenok was using a patch file I just thought to do the same thing for sturgeon. It is indeed super nasty ;)

I will create new git branches and apply the needed patches on top of that. I can send you a list of needed branches and projects when I have verified that I have it all working on my own account.

Ideally it should be as generic as possible. ~~That may be the problem as I am using older lollipop files to get the graphics stack to work on marhsmallow.
(See current instructions for system tarbal creation for sturgeon here: https://github.com/MagneFire/meta-sturgeon-hybris/blob/master/README-system-dir.md)~~
Edit: Current builds seem to no longer require the lollipop files for the graphics stack.

MagneFire commented:
Alright, took me some time to figure out what patches were really needed, but here is a list of new projects:
android_frameworks_av (https://github.com/MagneFire/android_frameworks_av/tree/marshmallow-dr1.5-release)
android_system_media (https://github.com/MagneFire/android_system_media/tree/marshmallow-dr1.5-release)
android_frameworks_base (https://github.com/MagneFire/android_frameworks_base/tree/marshmallow-dr1.5-release)

And here are the new branches for the existing projects:
android_system_core (https://github.com/MagneFire/android_system_core/tree/marshmallow-dr1.5-release)
android_frameworks_native (https://github.com/MagneFire/android_frameworks_native/tree/marshmallow-dr1.5-release)
android_hardware_qcom_display (https://github.com/MagneFire/android_hardware_qcom_display/tree/marshmallow-dr1.5-release)
android_hardware_libhardware (https://github.com/MagneFire/android_hardware_libhardware/tree/marshmallow-dr1.5-release)

As you can see, most patches are directly copied from the lollipop branch, others are copied from the lenok.patch

FlorentRevest commented:
Thanks a lot. This is a lot easier to review and maintain. I updated/created the corresponding repositories. Please PR the manifests.

FlorentRevest commented:
I am sorry I shouldn't have closed this PR I thought it was somewhere else...
So please update the PR to use the upstream AsteroidOS repositories and also remove the commit that introduces a patch from the git history

MagneFire commented:
Glad to help :)
Can you also push the code to android_frameworks_base (https://github.com/AsteroidOS/android_frameworks_base)?

FlorentRevest commented:
It is being pushed right now but I have a flaky connection in the French
country side so I’ll just go to sleep hoping it’s pushed when I wake up...

Le sam. 18 janv. 2020 à 00:06, MagneFire <notifications@github.com> a
écrit :

> Glad to help :)
> Can you also push the code to android_frameworks_base (
> https://github.com/AsteroidOS/android_frameworks_base)?
>
> —
> You are receiving this because you modified the open/close state.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/AsteroidOS/android_manifest/pull/6?email_source=notifications&email_token=AAG5ZCEX225KFBHVL5GOIOLQ6I2XZA5CNFSM4KFY74GKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEJJHH3Y#issuecomment-575828975>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAG5ZCA3FO4BAUHIR5XXYH3Q6I2XZANCNFSM4KFY74GA>
> .
>

MagneFire commented:
Alright ;) Ill wait for that before pushing the adjusted manifest file.
I want to make sure that everything works.

MagneFire committed:
Adjust manifest for sturgeon.
Includes missing audio headers and apply some patches from the lollipop branches.
